js/views/scrollView.js

```diff
@@ -772,7 +773,21 @@ ionic.views.Scroll = ionic.views.View.inherit({
         }
       }
 
-      self.doTouchMove(getEventTouches(e), e.timeStamp, e.scale);
+      if(self.options.locking && self.__scrollLockFix.direction === null){
```

Comment:
You could swap `self.__scrollLockFix.direction === null` for `!self.__scrollLockFix.doFix` (slightly less verbose and ("nit-pickingly") slightly more performant).